packages/server-wallet/src/models/objective.ts

```diff
@@ -33,6 +33,7 @@ type ObjectiveStatus = 'pending' | 'approved' | 'rejected' | 'failed' | 'succeed
 type WalletObjective<O extends Objective> = O & {
   objectiveId: string;
   status: ObjectiveStatus;
+  waitingFor: string;
```

Comment:
~I don't like the idea of unconstrained strings ending up in a `text` column that stores finite state.~

e. I see it is updated to a constrained type in future work.